e2e/test/ConnectionStatusChangeHandlerTests.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.Devices.E2ETests
+{
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Exceptions;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Diagnostics.Tracing;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    [TestCategory("IoTHub-E2E")]
+    public class ConnectionStatusChangeHandlerTests : IDisposable
+    {
+        private readonly string DevicePrefix = $"E2E_{nameof(ConnectionStatusChangeHandlerTests)}";
+        private readonly string ModulePrefix = $"E2E_{nameof(ConnectionStatusChangeHandlerTests)}";
+        private static TestLogging _log = TestLogging.GetInstance();
+
+        private readonly ConsoleEventListener _listener;
+
+        public ConnectionStatusChangeHandlerTests()
+        {
+            _listener = TestConfig.StartEventListener();
+        }
+
+        [TestMethod]
+        public async Task DeviceClient_DeviceDeleted_Gives_ConnectionStatus_DeviceDisabled()
+        {
+            TestDevice testDevice = await TestDevice.GetTestDeviceAsync(DevicePrefix + "_DeviceClient").ConfigureAwait(false);
+            string deviceConnectionString = testDevice.ConnectionString;
+
+            var config = new Configuration.IoTHub.DeviceConnectionStringParser(deviceConnectionString);
+            string deviceId = config.DeviceID;
+
+            ConnectionStatus? status = null;
+            ConnectionStatusChangeReason? statusChangeReason = null;
+            bool deviceDisabledReceived = false;
+
+            using (DeviceClient deviceClient =  DeviceClient.CreateFromConnectionString(deviceConnectionString, Client.TransportType.Amqp_Tcp_Only))
+            {
```

Comment:
What about other protocols? AMQP web socket, MQTT, MQTT websocket and HTTP ?

Reply:
As discussed offline:

Tests for AMQPWS and AMQP have been added.

For MQTT, IoT Hub returns error code 0x05 (ConnectionRefused, not authorized) instead of something specific indicating that the device has been disabled/deleted. This translates to the `ConnectionStatusChangeReason` of `Bad_Credential` instead.

For HTTP, the ConnectionStatusChangeHandler isn't called so no test required